core/include/detray/tools/navigator.hpp

```diff
@@ -488,9 +533,36 @@ class navigator {
         return true;
     }
 
+    DETRAY_HOST_DEVICE
+    auto &get_detector() { return detector; }
+
+    DETRAY_HOST_DEVICE
+    const auto &get_detector() const { return detector; }
+
     private:
     /** the containers for all data */
     detector_t detector;
```

Comment:
Please, can somebody explain to me why this cannot not be e.g. a pointer? I am sure I am just missing something...

Reply:
Since that line was not modified, I didn't even look at it... :confused:

Right now the code only passes a navigator object to the CUDA kernel. (Well, "the data" of a navigator object.) And from that the device version of the navigator is built in the kernel.

If the navigator needs to refer to another entity to work, then the setup should be modified a bit. Which may actually be a pretty good thing to do... In that case the kernel should receive a data object describing the detector and not the navigator, build a "device detector" out of that data object, and then create a "device navigator" around that "device detector".

That could actually improve the design quite a bit. :thinking: But again, since this PR was not modifying this aspect of the code, it may be easier to handle in a separate modification after this has gone in.